README.md

```diff
@@ -1,32 +1,254 @@
+  --${PROTO_OUTPUT_LANG}_out=${GENERATED_PROTO_DIR} ${YOUR_PROTO_FILE}
+```
+
+### Generate grpc stubs
+
+Generating grpc stubs is language-specific.
+
+#### Java
+
+Find the path to the grpc plugin (gradle will pull in the dependency if you don't have it already):
+
+```
+./gradlew showGrpcJavaPluginPath
+```
+
+The command will print out the path to the executable. Hereafter, this will be referenced as `${GRPC_JAVA_PLUGIN}`.
+
+```
+protoc -I=${PROTOC_INCLUDE_DIR} -I=${GOOGLEAPIS_DIR} -I=${YOUR_PROTO_DIR} \
+  --plugin=protoc-gen-grpc=${GRPC_JAVA_PLUGIN} \
+  --grpc_out=${GENERATED_PROTO_DIR} ${YOUR_PROTO_FILE}
+```
+
+#### Other languages
+
+TODO
+
+### Generate initial client config
+
+You need to locate/decide on the following before you call config generation:
+
+1. The service yaml file, hereafter referenced as `${YOUR_SERVICE_YAML}`.
+2. The file name of your output client config file. Hereafter, this will be referenced as
+   `${YOUR_CLIENT_CONFIG}`
+
+Run the following command to generate the client config:
+
+```
+cd ${TOOLKIT_DIR}
+java -cp build/libs/gapic-generator-*-fatjar.jar com.google.api.codegen.configgen.ConfigGeneratorTool \
+  --descriptor_set=${YOUR_DESCRIPTOR_FILE} --service_yaml=${YOUR_SERVICE_YAML} \
+  -o=${YOUR_CLIENT_CONFIG}
+```
+
+You can safely ignore the warning about control-presence.
+
+### Manually tweak the generated client config
+
+The generated client config contains FIXME comments with instructions on how to choose values
+in the client config. The client config should work as-is though; tweaks are only necessary to improve
+the quality of the generated output.
+
+### Create a package metadata config file
+
+You need to locate/decide on the following before you create the package metadata config file:
+
+1. The output file to contain the package metadata for your client generation. Hereafter, this will be
+   referenced as `${PKG_META_CONFIG}`.
+2. A short name, which typically does not include the major version. For example, Cloud Language v1's short name
+   is `language`. Hereafter, this will be referenced as `${SHORT_NAME}`.
+3. The major version. The first major version will typically be `v1`. Hereafter, this will be referenced as
+   `${MAJOR_VERSION}`.
+4. The package name, which typically includes the fully-qualified product name and major version.
+   For example, Cloud Language v1's package name is `google-cloud-language-v1`. Hereafter, this will be
+   referenced as `${PACKAGE_NAME}`.
+5. The proto path, which is the proto's package with dots converted to slashes. For example, Cloud Language v1's
+   proto path is `google/cloud/language/v1`. Hereafter, this will be referenced as `${PROTO_PATH}`.
+
+First, copy the contents of api_defaults.yaml and dependencies.yaml into a single file, hereafter
+referenced as `${PKG_META_CONFIG}`.
+
+```
+cp ${GOOGLEAPIS_DIR}/gapic/packaging/api_defaults.yaml ${PKG_META_CONFIG}
+cat < ${GOOGLEAPIS_DIR}/gapic/packaging/dependencies.yaml >> ${PKG_META_CONFIG}
```

Comment:
Wow... it looks like we are really not ready to distribute gapic-gen as a separate tool

Reply:
Nope! That is actually the key point of this whole exercise, to bring the mess out in the clear, so that we know what to clean up.